WebContent/bank/transaction.jsp

```diff
@@ -0,0 +1,162 @@
+				if (myform.startDate.value != ""){
+					var valid = false;
+					var splitStrings = myform.startDate.value.split("-");
+					if (splitStrings.length == 3) {
+						var year = parseInt(splitStrings[0]);
+						var month = parseInt((splitStrings[1].charAt(0)==0 && splitStrings[1].length == 2)?splitStrings[1].charAt(1):splitStrings[1]);
+						var day = parseInt((splitStrings[2].charAt(0)==0 && splitStrings[2].length == 2)?splitStrings[2].charAt(1):splitStrings[2]);
+
+						var validNums = !(isNaN(year) || isNaN(month) || isNaN(day));
+						
+						if (validNums)
+							valid = validateDate(month, day, year);
+					}
+			
+					if (!valid){
+						alert ("'After' date of " + myform.startDate.value + " is not valid.");
+						return false;
+					}
+				}
+			
+				if (myform.endDate.value != ""){
+					var valid2 = false;
+					var splitStrings2 = myform.endDate.value.split("-");
+					if (splitStrings2.length == 3) {
+						var year2 = parseInt(splitStrings2[0]);
+						var month2 = parseInt((splitStrings2[1].charAt(0)==0 && splitStrings2[1].length == 2)?splitStrings2[1].charAt(1):splitStrings2[1]);
+						var day2 = parseInt((splitStrings2[2].charAt(0)==0 && splitStrings2[2].length == 2)?splitStrings2[2].charAt(1):splitStrings2[2]);
+			
+						var validNums2 = !(isNaN(year2) || isNaN(month2) || isNaN(day2));
+						
+						if (validNums2)
+							valid2 = validateDate(month2, day2, year2);
+					}
+					
+					if (!valid2){
+						alert ("'Before' date of " + myform.endDate.value + " is not valid.");
+						return false;
+					}
+				}
+				return true;
+			}
+			
+			function validateDate(month, day, year){
+				try {
+					var thisDate = new Date();
+					var wrongMonth = month<1 || month>12;
+					var wrongDay = (day<1) || (day>31) || (day>30 && ((month==4)||(month==6)||(month==9)||(month==11))) || (day>29 && month==2 && (year%4==0) && (year%100!=0 || year%400==0)) || (day>28 && month==2 && ((year%4!=0) || (year%100==0 && year%400!=0))); 
+					var wrongYear = year < 1990 || year > parseInt(thisDate.getFullYear());
+			
+					var thisYear = parseInt(thisDate.getFullYear());
+					var thisMonth = parseInt(thisDate.getMonth())+1;
+					var thisDay = parseInt(thisDate.getDate());
+					var wrongDate = year==thisYear && ((thisMonth<month) || (thisMonth==month && thisDay<(day-1)));
+			
+					if (wrongMonth ||wrongDay || wrongYear || wrongDate)
+						return false;
+					
+					} catch (error){
+						return false;
+					}
+					
+					return true;
+			}
+		</script>
+		
+		<font style="bold" color="red"><%=error%></font>
+		<form id="Form1" name="Form1" method="post" action="showTransactions" onsubmit="return (confirminput(Form1));">
+		<table border="0" style="padding-bottom:10px;">
+		    <tr>
+		        <td valign=top>After</td>
+		        <td><input id="startDate" name="startDate" type="text" value="<%=(request.getParameter("startDate")==null)?"":request.getParameter("startDate")%>"/><br /><span class="credit">yyyy-mm-dd</span></td>
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from `an HTTP parameter` flows into `print`, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

<h6>
Line 126 | <a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 842</a> | <a href="https://learn.snyk.io/lesson/xss/">Learn more about this vulnerability</h6>

<details>

<summary><b>Data flow: 4 steps</b></summary>

`Step 1 - 4` https://github.com/snykyzach/zach-goof/blob/f0d3e6e422831193b61ae2b06af494a79d37a464/WebContent/bank/transaction.jsp#L126

</details>